### PR TITLE
Validating the form

### PR DESCRIPTION
**Pull request by fowobi**

### CODE REVIEW

File: client/src/components/ShowResult.jsx [hunk 1]
```diff
@@ -1,15 +1,31 @@
 import React, { useState, useEffect } from "react";
 import TeacherOverride from "./TeacherOverride";
+import "../styles/ShowResult.css";
 
 function ShowResult({ selectedAnswers, questions, comments = [],teacherName, pupilName,date }) {
```
Comment (alinosratipour) [inline client/src/components/ShowResult.jsx:5]: I think date props in unused

File: client/src/components/ShowResult.jsx [hunk 2]
```diff
@@ -1,15 +1,31 @@
...
   const [showResults, setShowResults] = useState(false);
   const [totalScore, setTotalScore] = useState(0);
   const [overrideScore, setOverrideScore] = useState("");
+  const [incompleteQuestionIndex, setIncompleteQuestionIndex] = useState(-1);
+
   const handleShowResults = () => {
```
Comment (alinosratipour) [inline client/src/components/ShowResult.jsx:11]: I am not sure what you want to achive here but inseted of for loop statment you be able to use this
```
   const handleShowResults = () => {
   const incompleteIndex = questions.findIndex((_, i) => {
    const answerId = selectedAnswers[i];
    const comment = comments[i];
    return !answerId || !comment.trim();
  });
  
```

Comment (fowobi) [reply client/src/components/ShowResult.jsx:11]: thank you for you feedback,I have made the neccessary correction

File: client/src/components/ShowResult.jsx (hunk 1 above)
Comment (fowobi) [reply client/src/components/ShowResult.jsx:5]: Yes, it is another person that is handle it,the person will fix it.thank you

**Pull request merged**